net-misc/olsrd/files/olsrd-0.9.8-clang16-buildfix.patch

```diff
@@ -0,0 +1,46 @@
+https://github.com/OLSR/olsrd/pull/124
+From: Brahmajit Das <brahmajit.xyz@gmail.com>
+Date: Wed, 31 May 2023 13:24:17 +0530
+Subject: [PATCH] fix build error with clang16
+
+Signed-off-by: Brahmajit Das <brahmajit.xyz@gmail.com>
+--- a/lib/filtergw/src/olsrd_filtergw.c
++++ b/lib/filtergw/src/olsrd_filtergw.c
+@@ -54,6 +54,7 @@
+ #include "log.h"
+ #include "routing_table.h"
+ #include "olsr_cfg.h"
++#include "parser.h"
+ 
+ #include <stdio.h>
+ #include <string.h>
+@@ -80,6 +81,10 @@ struct filter_group {
+ 
+ static struct filter_group * filter_groups = NULL;
+ 
++// Delcaring function prototype here, fixes build error with clang-16
++// Refer: https://bugs.gentoo.org/898090
++int should_filter(union olsr_ip_addr * originator);
```

Comment:
But why? This needs more justification.

Keep in mind that for _all_ of these bugs, there's always a wrong fix and a right fix:
- a cast (sometimes it's appropriate, but keep in mind, it's just effectively silencing the warning)
- passing the right argument (which may involve a cast, but not always, but a cast should not be done blindly)

Reply:
Ah, I see _this_ part (the redeclaration) is for an implicit. func. decl error, which is OK. But my above point still stands wrt casts and fixing these bugs in general.

Reply:
I don't think this was related to -Wincompatible-function-pointer-types, but rather -Werror=implicit-function-declaration

Reply:
> Doesn't compile with clang 16 on musl.

Thanks for the report, I'll fix it (don't have access to musl system right now).